prometheus_es_exporter/__init__.py

```diff
@@ -110,13 +113,15 @@ def gauge_generator(metrics):
         yield gauge
 
 
-def run_query(es_client, name, indices, query, timeout):
+def run_query(es_client, name, indices, query, timeout, zero_query_gauges_on_exception):
     try:
         response = es_client.search(index=indices, body=query, request_timeout=timeout)
 
         metrics = parse_response(response, [name])
     except Exception:
         logging.exception('Error while querying indices [%s], query [%s].', indices, query)
+        if zero_query_gauges_on_exception:
```

Comment:
Will this zero only query-related gauges or all? If all - I'd suggest making `name` input for  `zero_gauges()`

Reply:
We'll need to only zero gauges created by the query we're currently running, not all gauges.

If we pass in the `name`, `zero_gauges()` could filter the gauges by metric name. However, that won't work correctly when one query's name is the same as the first part of another query. We probably need to restructure the `gauges` global, so the gauges are grouped by query name, not just metric name.